cdap-runtime-ext-emr/src/main/java/co/cask/cdap/runtime/spi/provisioner/emr/EMRClient.java

```diff
@@ -89,7 +89,8 @@ public String createCluster(String name) {
 
     RunJobFlowRequest request = new RunJobFlowRequest()
       .withName(name)
-      .withReleaseLabel(emrConf.getReleaseLabel())
+      // all 4.9.x is java 7... which we don't support, so EMR 5.0.0 is our minimum
```

Comment:
5.15 uses spark 2.3, do our spark pipelines work?

Reply:
emr 5.0.0 uses spark2, so it seems like we can't support spark1 programs. We should throw an exception if spark compat is spark1.

Reply:
I've tried it out, and Spark execution engine doesn't work with either EMR verison (5.0.0 or 5.15.0).
I'm leaning towards implementing Spark support in a separate PR; filing the JIRA for now: https://issues.cask.co/browse/CDAP-13818.

Reply:
do we know what the issue is? that jira is pretty bare

Reply:
oh was it due to running it from your IDE?

Reply:
I spoke with Terence and learned that even with Dataproc, if you run CDAP from your IDE, you won't be able to run a pipeline against a cloud runtime if the execution engine is Spark.

I'm building the sandbox distribution and then I'll test Spark against EMR there and follow up with more details.

Reply:
It worked with the sandbox distribution.